src/reducers/JsonApiReducer.ts

```diff
@@ -136,6 +143,28 @@ export class JsonApiReducer {
         );
     }
 
+    private getEntitiesWithoutDeletedItem(
+        entities: Entities,
+        endpoint: string,
+    ): Entities {
+        // endpoint is of the following format: "/cars/2"
+        const keyToDelete = endpoint.split("/")[1];
+        const idToDelete = endpoint.split("/")[2];
+
+        // Removes entity (object of objects, cars:{1:{...},2:{...}} element from original
+        const { [keyToDelete]: parentValue, ...noChild } = entities;
+
+        // Removes the request-deleted item from object of objects extracted above
+        // @ts-ignore
+        const { [idToDelete]: removedValue, ...childWithout } = parentValue;
+
+        // Merge back together
+        return {
+            ...noChild,
+            [keyToDelete]: childWithout,
+        };
```

Comment:
Looks nice!

I'd suggest a minor improvement in naming so it's easier to understand on the last return what actually happened. 
Also, fixed ts ignore with casting to the required type. It was necessary because it could not index type "undefined", which was a possible scenario. That should return initial state.

```suggestion
// endpoint is of the following format: "/cars/2"
        const resourceType = endpoint.split("/")[1];
        const recordId = endpoint.split("/")[2];
        const {
            [resourceType]: targetResource,
            ...entitiesWithoutTargetResource
        } = entities;

        /**
         * nothing to delete, returning initial state
         */
        if (targetResource && targetResource[recordId] === undefined) {
            return entities;
        }

        // Removes the request-deleted item from object of objects extracted above
        const {
            [recordId]: removedValue,
            ...targetResourceWithoutDeletedKey
            // casting to identifiers because of the check for undefined in the earlier conditional
        } = targetResource as Identifiers;

        // Merge back together
        return {
            ...entitiesWithoutTargetResource,
            [resourceType]: targetResourceWithoutDeletedKey,
        };
```